state/factory/workingset.go

```diff
@@ -404,6 +404,40 @@ func (ws *workingSet) process(ctx context.Context, actions []action.SealedEnvelo
 	return ws.finalize()
 }
 
+func (ws *workingSet) validateSystemActionLayout(ctx context.Context, actions []action.SealedEnvelope) error {
+	reg := protocol.MustGetRegistry(ctx)
+	postSystemActions := []action.Envelope{}
+	for _, p := range reg.All() {
+		if psc, ok := p.(protocol.PostSystemActionsCreator); ok {
+			elps, err := psc.CreatePostSystemActions(ctx, ws)
```

Comment:
IMO, it's inefficient to regenerate the post-sys actions for each incoming block for reference. The actions should be generated and cached in the factory and be passed into `ValidateBlock()`.

Reply:
> IMO, it's inefficient to regenerate the post-sys actions for each incoming block for reference. The actions should be generated and cached in the factory and be passed into `ValidateBlock()`.

The regeneration should be necessary for different heights. Is there a scenario where multiple blocks need to be verified at the same height?

Reply:
ok. it's a corner case. nervermind